### DISCUSSION TITLE
Failure to create a box

### DISCUSSION THREAD

**Discussion by TakaYM** [ansys/pyansys-geometry#972] [Q&A] [answered]
I am trying Pygeometry and launched SpaceClaim with the following script.
~~~
_dect ={"length":"22"} 
modeler = launch_modeler_with_spaceclaim()
output = modeler.run_discovery_script_file("D:\test.scscript",script_args = _dect )
~~~

 However, when running to create a box shape with the subsequent script, it did not run. 
~~~
import time
val = float(_dect ["length"])
startPoint = Point.Create(MM(0), MM(0), MM(0)) 
endPoint = Point.Create(MM(val), MM(val), MM(val)) 
BlockBody.Create(startPoint, endPoint) 

time.sleep(5)

print "fin"
~~~

Could you please provide advice on what might be wrong with the above scripts?

Thanks in advance.
Takahiro Yamaguchi

**Comment by RobPasMue** [comment]
Hi @TakaYM, thanks for opening this issue! I have several follow up questions.

* What version of Ansys are you using (24R1, 23R2...)?
* Could you share as well the output of ``pip freeze`` on your Python environment?
* Can you show the tracestack from running your program?
* Once you run ``launch_modeler...``, can you do a ``print(modeler)`` and show us the output?

Also, quick guess on my side. It might be a typo, but there should be no space between _dect and ["length"] on your ".scscript". See current line 2 of the script you posted.

Thanks in advance!

**Comment by TakaYM** [comment]
Hi @RobPasMue - I asked the customer whether solved problem after deleting space between _dect and ["length"]. It seems that they could not  not solved this problem. Also, the customer answered to your question below.

- What version of Ansys are you using (24R1, 23R2...)?
They are using 23R2.
-Could you share as well the output of pip freeze on your Python environment?
I'm waiting answer from the customer.
- Can you show the tracestack from running your program?
![11111](https://github.com/ansys/pyansys-geometry/assets/158134249/294f1098-bb19-4a80-b4ba-4d4d4d1b3cc3)
- Once you run launch_modeler..., can you do a print(modeler) and show us the output?
Ansys Geometry Modeler (0x1975bade800)
Ansys Geometry Modeler Client (0x1975ba9a440)
 Target:   localhost:60660
 Connection: Healthy

And, I'd like to share the customer's python file and scscript file via onedrive.
https://ansys-my.sharepoint.com/:f:/p/takahiro_yamaguchi/Ev_Q3psbJS9AvKF-pKzZi4YBZUL7hErtvDbYe2gxAX8Omg?e=7g0uOQ

Regrads,
Takahiro Yamaguchi

**Reply by RobPasMue** [reply]
Hi @TakaYM!

Okay according to question 1... if they are using 23R2, they should use version ``0.3.2`` of PyAnsys Geometry. Version ``0.4.X`` will have breaking changes for them.

I am running SpaceClaim 24R1 with your current script. Launching SpaceClaim works fine. Executing the script failed. I had a look at the script and found the error...

Script arguments are passed as a dictionary called "argsDict" inside SpaceClaim. Meaning that the Python dictionary that you would get inside SpaceClaim is

```py
argsDict = {"length": "20"}
```

Consequently, you need to adapt your SpaceClaim to use this dictionary instead of ``_dect``

```py
import time
val = float(argsDict["length"])
startPoint = Point.Create(MM(0), MM(0), MM(0)) 
endPoint = Point.Create(MM(val), MM(val), MM(val)) 
BlockBody.Create(startPoint, endPoint) 

time.sleep(5)

print "fin"
```

Hope this helps - if so, mark this answer as the correct answer in GitHub. Really appreciated @TakaYM!

**Reply by RobPasMue** [reply, marked as answer]
We added this PR with enhanced documentation for your specific question:

https://github.com/ansys/pyansys-geometry/pull/977
